infra/src/main/kotlin/fi/fta/geoviite/infra/linking/LinkingService.kt

```diff
@@ -216,6 +216,16 @@ class LinkingService @Autowired constructor(
         return linkingDao.fetchPlanLinkStatus(planId = planId, publishType = publishType)
     }
 
+    fun getGeometryPlanLinkStatuses(
+        planIds: List<IntId<GeometryPlan>>,
+        publishType: PublishType,
+    ): List<GeometryPlanLinkStatus> {
+        logger.serviceCall(
+            "getGeometryPlanLinkStatuses", "planIds" to planIds, "publishType" to publishType
+        )
+        return planIds.map { planId -> linkingDao.fetchPlanLinkStatus(planId = planId, publishType = publishType) }
```

Comment:
Voi olla että varsinkin tälle olisi tehtävissä jonkinlainen järkevämpikin massahaku. En tosin lähtenyt toteuttamaan sitä tässä, sillä kokonaisuudessaan ajassa mitattuna linkingstatusten hakeminen on suunnitelmien layout-hakuun verrattuna a drop in the bucket